allennlp/training/callbacks/callbacks.py

```diff
@@ -0,0 +1,512 @@
+
+            is_best_so_far = training_states.pop("is_best_so_far", True)
+            self.checkpointer.save_checkpoint(
+                    model_state=trainer.model.state_dict(),
+                    epoch=trainer.checkpoint_epoch,
+                    training_states=training_states,
+                    is_best_so_far=is_best_so_far)
+
+
+        elif event == Events.RESTORE_CHECKPOINT:
+            # Restores the model and training state from the last saved checkpoint.
+            # This includes an epoch count and optimizer state, which is serialized separately
+            # from model parameters. This function should only be used to continue training -
+            # if you wish to load a model for inference/load parts of a model into a new
+            # computation graph, you should use the native Pytorch functions:
+            # `` model.load_state_dict(torch.load("/path/to/model/weights.th"))``
+
+            # If ``self._serialization_dir`` does not exist or does not contain any checkpointed weights,
+            # this will do nothing.
+            try:
+                model_state, training_state = self.checkpointer.restore_checkpoint()
+            except RuntimeError:
+                traceback.print_exc()
+                raise ConfigurationError("Could not recover training from the checkpoint.  "
+                                         "Did you mean to output to a different serialization directory "
+                                         "or delete the existing serialization directory?")
+
+            if not training_state:
+                # No checkpoint to restore, start at 0
+                trainer.epoch_number = 0
+                return
+
+            trainer.model.load_state_dict(model_state)
+
+            # Restore state_dict attrs
+            for attr in self.state_dict_attrs:
+                state_attr = getattr(trainer, attr)
+                if state_attr is not None:
+                    state_attr.load_state_dict(training_state[attr])
+
+            # Restore other attrs
+            for attr in self.other_attrs:
+                setattr(trainer, attr, training_state[attr])
+
+            # Restore callback attrs
+            for callback in trainer.handler.callbacks:
+                callback.restore_training_state(training_state)
+
+            if isinstance(training_state["epoch"], int):
+                trainer.epoch_number = training_state["epoch"] + 1
+            else:
+                trainer.epoch_number = int(training_state["epoch"].split('.')[0]) + 1
+
+        elif event == Events.TRAINING_END:
+            # Load the best model state before returning
+            best_model_state = self.checkpointer.best_model_state()
+            if best_model_state:
+                trainer.model.load_state_dict(best_model_state)
+
+    @classmethod
+    def from_params(cls, params: Params, serialization_dir: str) -> 'CheckpointCallback':  # type: ignore
```

Comment:
Same here; can't we get rid of this?  Maybe by changing the way `Checkpointer.from_params` works somehow?